src/Math-Tests-Complex/PMComplexTest.class.st

```diff
@@ -783,6 +729,72 @@ PMComplexTest >> testTanh [
 	self assert: (c2 imaginary closeTo: c tanh imaginary).
 ]
 
+{ #category : #'testing - complex conjugation' }
+PMComplexTest >> testThatComplexConjugateHasEqualRealPartAndAnImaginaryPartWithEqualMagnitudeButOppositeSign [
+	| z |
+	z := 3 + 2 i.
+	
+	self assert: z complexConjugate equals: 3 - 2 i
+]
+
+{ #category : #'testing - complex conjugation' }
+PMComplexTest >> testThatComplexConjugateOfAPureRealNumberIsItself [
+	| z |
+	z := 5 + 0 i.
+	
+	self assert: z complexConjugate equals: 5.
+]
+
+{ #category : #'testing - complex conjugation' }
+PMComplexTest >> testThatComplexConjugationIsAnInvolution [		
+	| z |
+	z := -9 + 24 i.
+	
+	self assert: z complexConjugate complexConjugate equals: -9 + 24 i
+]
+
+{ #category : #'testing - complex conjugation' }
+PMComplexTest >> testThatComplexConjugationIsDistributiveOverAddition [
+	| z w |
+	z :=  -3 + 6 i.
+	w := 10 - 4 i.
+	
+	self assert: (z + w) complexConjugate equals: (z complexConjugate + w complexConjugate)
+]
+
+{ #category : #'testing - complex conjugation' }
+PMComplexTest >> testThatComplexConjugationIsDistributiveOverDivision [
+	| z w |
+	z := 6 - 15 i.
+	w := 10 + 5 i.
+	
+	self assert: (z / w) complexConjugate equals: (z complexConjugate / w complexConjugate)
+]
+
+{ #category : #'testing - complex conjugation' }
+PMComplexTest >> testThatComplexConjugationIsNotSymmetric [
+	| z |
+	z := 3 - 2 i.
+	
+	self deny: z complexConjugate equals: z
+]
+
+{ #category : #'testing - complex conjugation' }
+PMComplexTest >> testThatComplexConjugationPreservesModulus [
+	| z |
+	z := 3 - 4 i.
+	
+	self assert: z complexConjugate abs equals: 5
+]
+
+{ #category : #'testing - complex conjugation' }
+PMComplexTest >> testThatTwoEntirelyDifferentComplexNumbersAreNotComplexConjugatesOfEachOther [
+	| z |
+	z := -0.5 - 1 i.
+	
+	self deny: z complexConjugate equals: (3 - 2i)
```

Comment:
I would remove ALL those tests.
They pass **by definition** if `testComplexConjugate` passes

There is no need to pollute the code with 6 tests that can not possibly fail
The only way this can fail is either:

(a) if `testComplexConjugate` fails
(b) if complex numbers don't behave as expected, but this should be (an it is) reflected in the tests of complex numbers. It has nothing to do with complex conjugates specifically